tests/test_finder.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Red Hat, Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+import os
+from argparse import Namespace
+from unittest import TestCase
+from unittest.mock import patch, call
+from tempfile import TemporaryDirectory
+
+
+from znoyder.finder import find_jobs
+from znoyder.finder import find_triggers
+from znoyder.finder import find_templates
+from znoyder.finder import main
+from znoyder.lib.exceptions import TriggerTypeError
+from znoyder.lib.exceptions import PathError
+from znoyder.lib.zuul import JobTriggerType
+
+
+class TestFinder(TestCase):
+    """Test the downloader module."""
+
+    def setUp(self):
+        self.test_directory = TemporaryDirectory()
+        self.dest_dir = self.test_directory.name
+        self.file_name = os.path.join(self.dest_dir, "zuul.d")
+        with open(self.file_name, "w", encoding="utf-8") as fw:
+            fw.write("- project:\n")
+            fw.write("    templates:\n")
+            fw.write("      - template1\n")
+            fw.write("      - template2\n")
+            fw.write("    check:\n")
+            fw.write("      jobs:\n")
+            fw.write("        - job1\n")
+            fw.write("        - job2\n")
+            fw.write("\n")
+            fw.write("- project-template:\n")
+            fw.write("    template1:\n")
+            fw.write("    name: template1\n")
+            fw.write("    check:\n")
+            fw.write("      jobs:\n")
+            fw.write("        - job1\n")
+            fw.write("        - job2\n")
+            fw.write("- project-template:\n")
+            fw.write("    template2:\n")
+            fw.write("    name: template2\n")
+            fw.write("    check:\n")
+            fw.write("      jobs:\n")
+            fw.write("        - job1\n")
+            fw.write("        - job2\n")
+
+    def tearDown(self):
+        self.test_directory.cleanup()
+
+    def test_find_triggers(self):
+        """Tests taht find_triggers works correctly."""
+        triggers = "experimental,post,gate,check,templates"
+        result = find_triggers(triggers)
+        expected = [4, 3, 2, 1, 0]
```

Comment:
I remember this is defined that silly way in code. Do you have any idea we can do something better here? JobTriggerType is static, so maybe we can just get this mapping somehow? Or it would not make sense, what do you think?

Reply:
You mean to use in the test, or for general use in znoyder? For the test it would be more robust to use the JobTriggerType, I'll change it